trainee_template/hyperparameters.amlg

```diff
@@ -294,17 +309,32 @@
 							(assoc deviations_map (get (current_value 1) "featureDeviations") )
 
 							(map
-								(lambda
-									(assign (assoc
-										deviations_map
-											(set
-												deviations_map
-												(current_index 1)
-												;max of user specified and stored deviation in the map
-												(max (current_value 1) (get deviations_map (current_index 1)) )
+								(lambda (let
+									(assoc
+										deviation (get deviations_map (current_index 1))
+										user_error_value (current_value 1)
+										feature (current_index 1)
+									)
+
+									(declare (assoc
+										deviation_value
+											;set to max of user specified error and stored deviation in the map
+											(if (!= (null) deviation)
+												;if deviation is a tuple, update the first value
+												(if (= (list) (get_type deviation))
+													(set deviation 0 (max user_error_value (first deviation)) )
+													;else just update the value itself
+													(max user_error_value deviation)
+												)
 											)
 									))
-								)
+
+									(if (!= (null) deviation_value)
```

Comment:
and what if deviation_value is null? If that's possible, should the deviation value jus tbe the user_error_value? If so, I don't see that represented here

Reply:
if it's null that means we are not using deviations, thus we wouldn't want to specify a value to make sure it stays null.  if it's not null, either it's a value or a tupel with null accuracies, and in both cases the actual deviation would be ovewritten with the user_error_value in the code block above (lines 320-330)